pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -334,6 +324,43 @@ func GatherContainerImages(i *Gatherer) func() ([]record.Record, []error) {
 			continueValue = pods.Continue
 		}
 
+		// Transform map into a list for sorting.
+		imageCounts := []tmpImageCountEntry{}
+		for img, countMap := range img2month2count {
+			totalCount := 0
+			for _, count := range countMap {
+				totalCount += count
+			}
+			imageCounts = append(imageCounts, tmpImageCountEntry{
+				Image:         img,
+				TotalCount:    totalCount,
+				CountPerMonth: countMap,
+			})
+		}
+
+		// Sort images from most common to least common.
+		sort.Slice(imageCounts, func(i, j int) bool {
```

Comment:
It would be interesting to see the a sample of the data here, because I suspect the most common containers will be end-user application containers rather than the RH shipped containers which we are interested in writing rules for. If that sample shows that they are end-user application containers then we could perhaps inverse this. However better again would be to get the currently running containers in my opinion.

Reply:
Ok, my mistake, seems you are actually getting the current running containers. If that still exceeds the max image count on target clusters than we probably need to re-think this PR and stick with the current method of getting the image ref from the pod spec.

An alternative might be to lookup the container in Pyxis using the sha, and then use the returned data to filter based on image ref or build labels from there. It depends what's more important, time to gather data, or the size of data transmitted for post-processing. If time to gather is not an issue, using Pyxis to refine the collected data would seems like a valid option to me.

Reply:
As mentioned in my email, we have considered this, but to avoid any unnecessary load, we are trying to keep the collection as simple as possible, at least for the moment being.

Reply:
Let's see if this method will get us the containers we need. If not, we can adjust the filtering later.